core/client/src/main/java/alluxio/client/block/stream/LocalFilePacketReader.java

```diff
@@ -53,6 +82,8 @@ public DataBuffer readPacket() {
     if (mPos >= mEnd) {
       return null;
     }
+    initIfNot();
```

Comment:
Consider putting init in the constructor? Otherwise the block is not guaranteed to be available even if reader is successfully created.